File: packages/ts/src/cli/config-paths-from-argv.ts
// The `-c`/`--config` flag is repeatable, so the scan collects every
// occurrence, in argv order.

/**
 * Every config value in argv, in order (`--config X`, `--config=X`, `-c X`,
 * `-c=X`, `-cX`), or the default when none are given.
 */
export function configPathsFromArgv(argv: string[]): string[] {
  const paths: string[] = [];
  let expectValue = false;
  for (const a of argv) {
    if (expectValue) {
      paths.push(a);
      expectValue = false;
    } else if (a === "--config" || a === "-c") {
      expectValue = true;
    } else if (a.startsWith("--config=")) {
      paths.push(a.slice("--config=".length));
    } else if (a.startsWith("-c")) {
      const value = a.slice("-c".length);
      paths.push(value.startsWith("=") ? value.slice("=".length) : value);
    }
  }
  if (expectValue) {
    // A bare trailing flag (no following value) yields "".
    paths.push("");
  }
  return paths.length > 0 ? paths : ["./.dirsql.toml"];
}

File: packages/python/dirsql/cli/config_paths_from_argv.py
"""Config-path extraction from the launcher's argv.

Every config flag occurrence counts -- ``-c``/``--config`` are repeatable,
and plugin discovery injects fragments as additional ``-c`` flags -- so the
scan collects them all, in argv order.
"""

from __future__ import annotations


def config_paths_from_argv(argv: list[str]) -> list[str]:
    """Every config value in argv, in order (``--config X``, ``--config=X``,
    ``-c X``, ``-c=X``, ``-cX``), or the default when none are given."""
    paths: list[str] = []
    i = 0
    while i < len(argv):
        a = argv[i]
        if a == "--config" or a == "-c":
            # A bare trailing flag (no following value) yields "".
            paths.append(next(iter(argv[i + 1 :]), ""))
            i += 2
            continue
        if a.startswith("--config="):
            paths.append(a[len("--config=") :])
        elif a.startswith("-c"):
            paths.append(a[len("-c") :].removeprefix("="))
        i += 1
    return paths or ["./.dirsql.toml"]
